### ISSUE TITLE
FREAK & BRISK Descriptors

### ISSUE THREAD

**Issue by mronian** [JuliaImages/ImageFeatures.jl#7]
FREAK Roadmap
- [x] Freak Best Pairs
- [x] Orientation Pairs
- [x] Mean Intensity
- [x] Orientation of Patch
- [x] Sample Pairs for Descriptor
- [ ] Modified Hamming Distance calculator (calculates distance in parts)

BRISK Roadmap
- [ ] Need AGAST corners
- [x] Orientation (Long & Short Pairs)
- [x] Sample Pairs
- [ ] Keypoint Maxima

**Comment by mronian** [comment]
@timholy Some descriptors like ORB and BRISK (currently working on this with FREAK) also calculate keypoints for description. If someone wants just the keypoints, there should be an option to retrieve those. But if the same keypoints are to be passed on to the `create_descriptor` function later on, they wont make any sense since we need the orientation as well as the scale of each keypoint. I was thinking of having a `Feature` type - 

``` jl
type Feature
     keypoint::Keypoint
     orientation::Float64
     scale::Float64
end
```

and then return an `Array{Feature}` from the `extract_features` function. Will such a structure make iteration over keypoints/features slow, since the person will probably just need the keypoints?

**Comment by timholy** [comment]
No, it should be fine. Make it an `immutable`, though; then it will be just 32 bytes per `Feature` in the array (2_Int+2_Float64=32 bytes).
